### ISSUE TITLE
bug: differing timeunit inequality matching fails 

### ISSUE THREAD

**Issue by drcapybara** [spaceandtimelabs/sxt-proof-of-sql#107]
Suppose I have a table of times denoted in seconds:

```rust
accessor.add_table(
    "sxt.table".parse().unwrap(),
    owned_table([timestamptz(
        "times",
        PoSQLTimeUnit::Second,
        PoSQLTimeZone::Utc,
        test_timestamps,
    )]),
    0,
);
```

Assume that the table contains the following:
```rust
vec![-1, 0, 1];
```

And I wish to query the following against this table:

```sql
"SELECT * FROM table WHERE times < timestamp '1970-01-01T00:00:00.000000001Z'"
```

One would expect that the query should give back [-1, 0], but instead, after parsing the query and generating a proof:

```rust
// Parse and execute the query
let query = QueryExpr::try_new(
    query_str.parse().unwrap(),
    "sxt".parse().unwrap(),
    &accessor,
)
.unwrap();

let proof = VerifiableQueryResult::<InnerProductProof>::new(query.proof_expr(), &accessor, &());

// Verify the results
let owned_table_result = proof
    .verify(query.proof_expr(), &accessor, &())
    .unwrap()
    .table;
```

we receive the following:

```bash
ProofError(VerificationError("sumcheck evaluation check failed"))
```

from the ```unwrap()``` on the ```verify()``` call:

```rust
    // Verify the results
    let owned_table_result = proof
        .verify(query.proof_expr(), &accessor, &())
        .unwrap()
```

Without a timeunit at all, this query is shown to succeed as expected in #75. I would assume that we should be able to compare times of differing units without incident so long as rounding does not occur, but in this case of inequality it should not.

Curiously, the query
```sql
"SELECT * FROM table WHERE times > timestamp '1970-01-01T00:00:00.000000001Z'"
```

succeeds without incident on a table of seconds, so perhaps this is a simple fix somewhere in the inequality expr node?

## Proofs-service
This issue produces buggy behavior in proofs-service because the gateway currently can only DDL tables into nanosecond precision, thus any query that is parsed as a timeunit differing from nanoseconds seems to be failing with a ```sumcheck verification failure```.

## Timeunit parsing:
For reference, timeunits are parsed here, I am still uneasy about this approach:
```rust
pub fn try_from(timestamp_str: &str) -> Result<Self, PoSQLTimestampError> {
    let dt = DateTime::parse_from_rfc3339(timestamp_str)
        .map_err(|e| PoSQLTimestampError::ParsingError(e.to_string()))?;

    let offset_seconds = dt.offset().local_minus_utc();
    let timezone = PoSQLTimeZone::from_offset(offset_seconds);
    let nanoseconds = dt.timestamp_subsec_nanos();
    let timeunit = if nanoseconds % 1_000 != 0 {
        PoSQLTimeUnit::Nanosecond
    } else if nanoseconds % 1_000_000 != 0 {
        PoSQLTimeUnit::Microsecond
    } else if nanoseconds % 1_000_000_000 != 0 {
        PoSQLTimeUnit::Millisecond
    } else {
        PoSQLTimeUnit::Second
    };

    Ok(PoSQLTimestamp {
        timestamp: dt.with_timezone(&Utc),
        timeunit,
        timezone,
    })
}
```

Perhaps there is a more sophisticated way of handling this? Because it seems like just checking the number of zeros at the end of a time is not always guaranteed to deliver the expected time unit for the given query.

**Comment by drcapybara** [comment]
The bug is [here](https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/644b0a784abb633f1824ddb73dd3d46c565ee19b/crates/proof-of-sql/src/sql/ast/inequality_expr.rs#L68):

```rust
let diff = if self.is_lte {
    scale_and_subtract(alloc, lhs_column, rhs_column, lhs_scale, rhs_scale, false)
        .expect("Failed to scale and subtract")
} else {
    scale_and_subtract(alloc, rhs_column, lhs_column, lhs_scale, rhs_scale, false)
        .expect("Failed to scale and subtract")
};
 ```
 
 rhs and lhs should be reversed in the gte branch. Additionally, there is a testing gap for >= / gte inequalities. This issue should be resolved in #110

**Comment by drcapybara** [comment]
closing this issue as resolved
